geopandas/geodataframe.py

```diff
@@ -898,15 +898,20 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
         if len(properties_cols) > 0:
             # convert to object to get python scalars.
             properties = self[properties_cols].astype(object).values
+            na_mask = pd.isna(properties)
```

Comment:
It might be more efficient to calculate the mask on the full dataframe (so it is still done on the columns' original dtype, instead of on the array that is coerced to object dtype). So something like `pd.isna(self[properties_cols]).values`